### PR TITLE
Grpc 4

### PR DESCRIPTION
**Pull request by oleastre**
This is some code I use while testing the grpc pricer.
It adds some way to debug the runner and avoid some errors when running without testpad.

### CODE REVIEW

File: foundations/src/main/java/io/yields/math/framework/data/Generator.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
     @Override
     default Collection<ExplorerListener> getListeners() {
-        return Collections.EMPTY_LIST;
```
Comment (jgheerardyn) [inline foundations/src/main/java/io/yields/math/framework/data/Generator.java]: Just out of curiosity: why do you prefer emptyList() over EMPTY_LIST?

File: testpad-core/src/main/java/io/yields/testpad/junit/TestDefinitionRunner.java [hunk 2]
```diff
@@ -196,6 +200,12 @@ public static void main(String... args) throws Exception {
             return;
         }
 
+        // If debug is enabled, enable slf4j debug logging
+        if(cmd.hasOption("debug")) {
```
Comment (jgheerardyn) [inline testpad-core/src/main/java/io/yields/testpad/junit/TestDefinitionRunner.java]: could you mention this command line arg in our wiki:
https://github.com/jgheerardyn/yields/wiki/TestDefinitionRunner
?

File: foundations/src/main/java/io/yields/math/framework/data/Generator.java (hunk 1 above)
Comment (oleastre) [inline foundations/src/main/java/io/yields/math/framework/data/Generator.java]: emptyList() returns you the EMPTY_LIST constant but casted to your variable, avoiding a warning.
For the compiler point of view, EMPTY_LIST does not have a type while emptyList does; so it's more type safe.

File: testpad-core/src/main/java/io/yields/testpad/junit/TestDefinitionRunner.java (hunk 2 above)
Comment (oleastre) [inline testpad-core/src/main/java/io/yields/testpad/junit/TestDefinitionRunner.java]: Done.

Comment (jgheerardyn) [inline testpad-core/src/main/java/io/yields/testpad/junit/TestDefinitionRunner.java]: thanks!

**Pull request merged**